### ISSUE TITLE
NestedCVWrapper leads to row misalignment

### ISSUE THREAD

**Issue by muellermarkus** [scikit-learn-contrib/category_encoders#526]
The way the NestedCVWrapper is implemented leads to a row misalignment. This is detrimental for downstream models. In the example below, encoded_train indices no longer align with y_train indices. 

```
import category_encoders as ce
import numpy as np
import pandas as pd
from category_encoders.wrapper import NestedCVWrapper
from sklearn.model_selection import KFold

rng = np.random.default_rng(42)
x = rng.integers(0, 5, size=(1000, 2))
y = rng.normal(size=(1000,))
df = pd.DataFrame({"x1": x[:, 0], "x2": x[:, 1], "y": y})
x_train = df[["x1", "x2"]].astype("category")
y_train = df["y"]

cv = KFold(n_splits=5, shuffle=True, random_state=42)
enc = ce.MEstimateEncoder(m=1)
enc = NestedCVWrapper(enc, cv=cv, random_state=42)
encoded_train = enc.fit_transform(x_train, y=y_train)

(x_train.index == encoded_train.index).all() # expect True
```
A quick fix would be to require shuffle=False. But this only holds for KFold. StratifiedKFold still suffers from row misalignment.

**Comment by wdm0006** [comment]
hey @muellermarkus thanks, taking a look
